src/JustSaying/Messaging/Channels/Multiplexer/RoundRobinQueueMultiplexer.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using JustSaying.Messaging.Channels.Context;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Channels.Multiplexer
+{
+    internal sealed class RoundRobinQueueMultiplexer : IMultiplexer, IDisposable
```

Comment:
I would like to see a simpler default multiplexer, along the lines of the `Merge<T>` method here: https://github.com/deniskyashif/blog/blob/master/content/posts/2019-12-08-csharp-channels-part-1.md
The round robin behaviour seems like a niche requirement (I could be missing some context), I'm suggesting we include it be default to a simpler one, thoughts?

Reply:
This is really interesting - we would essentially be running a task per reader than just waits for each to have a message ready and pumps it into the primary channel. 

I can't see any issue with that, for most situations its probably a cleaner default as we wouldn't be constantly enumerating the readers.

I'll put one together 👍

Reply:
Hmm, so that blog post leans pretty heavily on a `ReadAllAsync` which doesn't exist in the RTM channels API. I think we could replicate that behaviour in the same way the round robin multiplexer does by calling `WaitToReadAsync` and then writing to the channel in a loop.

Reply:
I've created a PR to add this, keen for your thoughts!
https://github.com/gkinsman/JustSaying/pull/45

Reply:
`ReadAllAsync` does exist in RTM, but if we are targeting .NET Framework then it won't be available, I guess the reasoning was that async streams are unsupported on .NET Framework ( it seems odd they chose to impact their API surface with this, but hey).

It's just:
```csharp
public virtual async IAsyncEnumerable<T> ReadAllAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
{
    while (await WaitToReadAsync(cancellationToken).ConfigureAwait(false))
    {
        while (TryRead(out T item))
        {
            yield return item;
        }
    }
}
```

Reply:
Should we not be using `IAsyncEnumerable<T>` at all then, as we're also targetting netfx users on net462? I know early on we avoided it, but it's creeped in now 🙂

Reply:
This is my stance; we can use it, and use libraries that use it. As soon as you want to do language level things (`await foreach` or `yield return` in an `async` block) then what you are doing is "unsupported", but still works (https://stu.dev/csharp8-doing-unsupported-things/). What makes this even more interested is that MS is encouraging people to do the unsupported thing with Nullable Reference Types, so there's a kinda "encouraged unsupported" thing going on 😆 

Replicating the `Merge<T>` approach, we'd have a local function returning `Task` (so we won't need to convert from `ValueTask` as we can just `await` them), and inline the `WaitToReadAsync` method essentially.

Reply:
I missed the comment regarding your PR, am looking now 👀

Reply:
Ah, yes I remember reading a bit about that when it happened. Do you think it's an attempt to be consistent in their guidance without having to provide support on a per-feature basis? Default interfaces are the big no-go, but as per your table it looks like basically everything else works with just types + compiler 🙂

Reply:
Yeah exactly. `IAsyncEnumerable<T>` is quite usable on older frameworks with `System.Interactive.Async`. Shameless self promotion, I cover it here:
https://youtu.be/xtskmbP93CA?t=1348

Reply:
Have now merged the PR to this branch (https://github.com/gkinsman/JustSaying/pull/45) that replaces the default RoundRobinQueueMultiplexer with a new MergingMultiplexer that works like this.